jdaviz/configs/mosviz/helper.py

```diff
@@ -46,89 +55,133 @@ def _extend_world(self, spec1d, ext):
+            return
+
+        self._panning_warning_triggered = False
+
+        idx_min = float((np.abs(world - min_1d)).argmin()) - extend_by
+        idx_max = float((np.abs(world - max_1d)).argmin()) - extend_by
+
+        self._update_in_progress = True
+        with self._scales2d.hold_sync():
+            self._scales2d.min = idx_min
+            self._scales2d.max = idx_max
+
+        self._update_in_progress = False
+
+    def _update_spec1d_x_axis(self, _):
         # This assumes the two spectrum viewers have the same x-axis shape and
         # wavelength solution, which should always hold
-        if change['old'] is None:
-            pass
-        else:
-            name = change['name']
-            if name not in ['min', 'max']:
-                return
-            new_idx = int(np.around(change['new']))
-            spec1d = self.app.get_viewer('table-viewer')._selected_data["spectrum-viewer"]
-            extend_by = int(self.app.data_collection[spec1d]["World 0"].shape[0])
-            world = self._extend_world(spec1d, extend_by)
-
-            scales = self.app.get_viewer('spectrum-viewer').scales
-            old_val = getattr(scales['x'], name)
-
-            # Warn the user if they've panned far enough away from the data
-            # that the viewers will desync
-            try:
-                val = world[new_idx+extend_by]
-            except IndexError:
-                val = old_val
-                msg = "Warning: panning too far away from the data may desync \
+        table_viewer = self.app.get_viewer('table-viewer')
+
+        if self._update_in_progress or table_viewer.row_selection_in_progress:
+            return
+
+        min_2d = self._scales2d.min
+        max_2d = self._scales2d.max
+
+        spec1d = table_viewer._selected_data["spectrum-viewer"]
+        extend_by = int(self.app.data_collection[spec1d]["World 0"].shape[0])
+        world = self._extend_world(spec1d, extend_by)
+
+        idx_min = int(np.around(min_2d)) + extend_by
+        idx_max = int(np.around(max_2d)) + extend_by
+
+        # Warn the user if they've panned far enough away from the data
+        # that the viewers will desync
+        if not (0 <= idx_min < len(world) and 0 <= idx_max < len(world)):
+            self._show_panning_warning()
+            return
+
+        self._panning_warning_triggered = False
+
+        self._update_in_progress = True
+        with self._scales1d.hold_sync():
+            self._scales1d.min = world[idx_min]
+            self._scales1d.max = world[idx_max]
+
+        self._update_in_progress = False
+
+    def _show_panning_warning(self):
+        now = time()
+        if not self._panning_warning_triggered and now > self._last_panning_warning + 5:
+            self._panning_warning_triggered = True
+            self._last_panning_warning = now
+            msg = "Warning: panning too far away from the data may desync \
                        the 1D and 2D spectrum viewers"
```

Comment:
```suggestion
            msg = ("Warning: panning too far away from the data may desync"
                   "the 1D and 2D spectrum viewers")
```

Reply:
Regarding line 133: I've moved these line and kept the formatting from the original code. Happy to change this, but I generally change as little as possible of the original code.